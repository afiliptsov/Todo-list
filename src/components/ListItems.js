import React from 'react'
import {TiDeleteOutline} from 'react-icons/ti'
import {BiCheckboxChecked, BiCheckbox} from 'react-icons/bi'
import '../styles/TodoListItem.css'
import PropTypes from 'prop-types'

const ListItems = props => {
  const {todoListItems, removeTodoItem, toggleItem} = props

  return (
    <div className="toDoListWrapper">
      {todoListItems.length ? (
        <ul>
          {todoListItems.map(activeCategory =>
            // Support for multiple active categories
            activeCategory.list.map(item => (
              <div className="toDoListItem" key={item.id}>
                <div
                  className="todoListItemIcon"
                  onClick={() => toggleItem(activeCategory.id, item)}
                >
                  {item.checked ? <BiCheckboxChecked /> : <BiCheckbox />}
                </div>
                <li>
                  <div
                    className="categoryToggle"
                    data-testid={item.description}
                  >
                    {item.description}
                  </div>
                </li>
                <div className="todoListItemIcon todoListItemDeleteIcon">
                  <TiDeleteOutline
                    onClick={() => removeTodoItem(activeCategory.id, item)}
                  />
                </div>
              </div>
            )),
          )}
        </ul>
      ) : (
        <div>Please add or select a Category</div>
      )}
    </div>
  )
}

//props type checking
ListItems.propTypes = {
  todoListItems: PropTypes.array,
  removeTodoItem: PropTypes.func,
  toggleItem: PropTypes.func,
}

export default ListItems
