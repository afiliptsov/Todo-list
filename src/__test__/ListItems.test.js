import {render, screen} from '@testing-library/react'
import ListItems from '../components/ListItems'
import mockState from './mockState'

describe('ListItems page elements are displayed', () => {
  it('renders List Items component from mock data', () => {
    render(<ListItems todoListItems={mockState} />)
    const getListItemElements = screen.getByTestId('tomatoes')
    expect(getListItemElements.innerHTML).toBe('tomatoes')
  })
})
