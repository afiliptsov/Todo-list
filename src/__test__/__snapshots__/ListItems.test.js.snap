// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list item snapshot validation list item shoulld match snapshot 1`] = `
<div
  className="toDoListWrapper"
>
  <ul>
    <div
      className="toDoListItem"
    >
      <div
        className="todoListItemIcon"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
          />
        </svg>
      </div>
      <div>
        <li>
          <div
            className="categoryToggle"
            data-testid="Hi there "
          >
            Hi there 
          </div>
        </li>
      </div>
      <div
        className="todoListItemIcon todoListItemDeleteIcon"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3c-4.963 0-9 4.038-9 9s4.037 9 9 9 9-4.038 9-9-4.037-9-9-9zm0 16c-3.859 0-7-3.14-7-7s3.141-7 7-7 7 3.14 7 7-3.141 7-7 7zM12.707 12l2.646-2.646c.194-.194.194-.512 0-.707-.195-.194-.513-.194-.707 0l-2.646 2.646-2.646-2.647c-.195-.194-.513-.194-.707 0-.195.195-.195.513 0 .707l2.646 2.647-2.646 2.646c-.195.195-.195.513 0 .707.097.098.225.147.353.147s.256-.049.354-.146l2.646-2.647 2.646 2.646c.098.098.226.147.354.147s.256-.049.354-.146c.194-.194.194-.512 0-.707l-2.647-2.647z"
          />
        </svg>
      </div>
    </div>
    <div
      className="toDoListItem"
    >
      <div
        className="todoListItemIcon"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
          />
          <path
            d="M10.996 12.556 9.7 11.285l-1.4 1.43 2.704 2.647 4.699-4.651-1.406-1.422z"
          />
        </svg>
      </div>
      <div>
        <li>
          <div
            className="categoryToggle"
            data-testid="<--- You can check items by clicking checkbox"
          >
            &lt;--- You can check items by clicking checkbox
          </div>
        </li>
      </div>
      <div
        className="todoListItemIcon todoListItemDeleteIcon"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3c-4.963 0-9 4.038-9 9s4.037 9 9 9 9-4.038 9-9-4.037-9-9-9zm0 16c-3.859 0-7-3.14-7-7s3.141-7 7-7 7 3.14 7 7-3.141 7-7 7zM12.707 12l2.646-2.646c.194-.194.194-.512 0-.707-.195-.194-.513-.194-.707 0l-2.646 2.646-2.646-2.647c-.195-.194-.513-.194-.707 0-.195.195-.195.513 0 .707l2.646 2.647-2.646 2.646c-.195.195-.195.513 0 .707.097.098.225.147.353.147s.256-.049.354-.146l2.646-2.647 2.646 2.646c.098.098.226.147.354.147s.256-.049.354-.146c.194-.194.194-.512 0-.707l-2.647-2.647z"
          />
        </svg>
      </div>
    </div>
    <div
      className="toDoListItem"
    >
      <div
        className="todoListItemIcon"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
          />
        </svg>
      </div>
      <div>
        <li>
          <div
            className="categoryToggle"
            data-testid="You can delete items too --->"
          >
            You can delete items too ---&gt;
          </div>
        </li>
      </div>
      <div
        className="todoListItemIcon todoListItemDeleteIcon"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3c-4.963 0-9 4.038-9 9s4.037 9 9 9 9-4.038 9-9-4.037-9-9-9zm0 16c-3.859 0-7-3.14-7-7s3.141-7 7-7 7 3.14 7 7-3.141 7-7 7zM12.707 12l2.646-2.646c.194-.194.194-.512 0-.707-.195-.194-.513-.194-.707 0l-2.646 2.646-2.646-2.647c-.195-.194-.513-.194-.707 0-.195.195-.195.513 0 .707l2.646 2.647-2.646 2.646c-.195.195-.195.513 0 .707.097.098.225.147.353.147s.256-.049.354-.146l2.646-2.647 2.646 2.646c.098.098.226.147.354.147s.256-.049.354-.146c.194-.194.194-.512 0-.707l-2.647-2.647z"
          />
        </svg>
      </div>
    </div>
    <div
      className="toDoListItem"
    >
      <div
        className="todoListItemIcon"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
          />
        </svg>
      </div>
      <div>
        <li>
          <div
            className="categoryToggle"
            data-testid="You can display multiple categories at the same time"
          >
            You can display multiple categories at the same time
          </div>
        </li>
      </div>
      <div
        className="todoListItemIcon todoListItemDeleteIcon"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3c-4.963 0-9 4.038-9 9s4.037 9 9 9 9-4.038 9-9-4.037-9-9-9zm0 16c-3.859 0-7-3.14-7-7s3.141-7 7-7 7 3.14 7 7-3.141 7-7 7zM12.707 12l2.646-2.646c.194-.194.194-.512 0-.707-.195-.194-.513-.194-.707 0l-2.646 2.646-2.646-2.647c-.195-.194-.513-.194-.707 0-.195.195-.195.513 0 .707l2.646 2.647-2.646 2.646c-.195.195-.195.513 0 .707.097.098.225.147.353.147s.256-.049.354-.146l2.646-2.647 2.646 2.646c.098.098.226.147.354.147s.256-.049.354-.146c.194-.194.194-.512 0-.707l-2.647-2.647z"
          />
        </svg>
      </div>
    </div>
    <div
      className="toDoListItem"
    >
      <div
        className="todoListItemIcon"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
          />
        </svg>
      </div>
      <div>
        <li>
          <div
            className="categoryToggle"
            data-testid="potatoes "
          >
            potatoes 
          </div>
        </li>
      </div>
      <div
        className="todoListItemIcon todoListItemDeleteIcon"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3c-4.963 0-9 4.038-9 9s4.037 9 9 9 9-4.038 9-9-4.037-9-9-9zm0 16c-3.859 0-7-3.14-7-7s3.141-7 7-7 7 3.14 7 7-3.141 7-7 7zM12.707 12l2.646-2.646c.194-.194.194-.512 0-.707-.195-.194-.513-.194-.707 0l-2.646 2.646-2.646-2.647c-.195-.194-.513-.194-.707 0-.195.195-.195.513 0 .707l2.646 2.647-2.646 2.646c-.195.195-.195.513 0 .707.097.098.225.147.353.147s.256-.049.354-.146l2.646-2.647 2.646 2.646c.098.098.226.147.354.147s.256-.049.354-.146c.194-.194.194-.512 0-.707l-2.647-2.647z"
          />
        </svg>
      </div>
    </div>
    <div
      className="toDoListItem"
    >
      <div
        className="todoListItemIcon"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 5c-1.103 0-2 .897-2 2v10c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2H7zm0 12V7h10l.002 10H7z"
          />
        </svg>
      </div>
      <div>
        <li>
          <div
            className="categoryToggle"
            data-testid="tomatoes"
          >
            tomatoes
          </div>
        </li>
      </div>
      <div
        className="todoListItemIcon todoListItemDeleteIcon"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3c-4.963 0-9 4.038-9 9s4.037 9 9 9 9-4.038 9-9-4.037-9-9-9zm0 16c-3.859 0-7-3.14-7-7s3.141-7 7-7 7 3.14 7 7-3.141 7-7 7zM12.707 12l2.646-2.646c.194-.194.194-.512 0-.707-.195-.194-.513-.194-.707 0l-2.646 2.646-2.646-2.647c-.195-.194-.513-.194-.707 0-.195.195-.195.513 0 .707l2.646 2.647-2.646 2.646c-.195.195-.195.513 0 .707.097.098.225.147.353.147s.256-.049.354-.146l2.646-2.647 2.646 2.646c.098.098.226.147.354.147s.256-.049.354-.146c.194-.194.194-.512 0-.707l-2.647-2.647z"
          />
        </svg>
      </div>
    </div>
  </ul>
</div>
`;
